tests/behavior/transforms/dedupe.toml

```diff
@@ -0,0 +1,118 @@
+[transforms.dedupe_simple]
+  inputs = []
+  type = "dedupe"
+  fields.match = ["a"]
+
+[[tests]]
+  name = "dedupe_simple"
+  [[tests.inputs]]
+    insert_at = "dedupe_simple"
+    type = "log"
+    [tests.inputs.log_fields]
+      a = 1
+      b = 2
+  [[tests.inputs]]
+    insert_at = "dedupe_simple"
+    type = "log"
+    [tests.inputs.log_fields]
+      a = 1
+      b = 3
+  [[tests.inputs]]
+    insert_at = "dedupe_simple"
+    type = "log"
+    [tests.inputs.log_fields]
+      a = 2
+      b = 4
+  [[tests.outputs]]
+    extract_from = "dedupe_simple"
+    [[tests.outputs.conditions]]
+      type = "check_fields"
+      "a.equals" = 1
+      "b.equals" = 2
+  [[tests.outputs]]
+    extract_from = "dedupe_simple"
+    [[tests.outputs.conditions]]
+      type = "check_fields"
+      "a.equals" = 2
+      "b.equals" = 4
+
+[transforms.dedupe_field_order]
+  inputs = []
+  type = "dedupe"
+  fields.match = ["a", "b"]
+
+[[tests]]
+  name = "dedupe_field_order"
+  [[tests.inputs]]
+    insert_at = "dedupe_field_order"
+    type = "log"
+    [tests.inputs.log_fields]
+      a = 1
+      b = 2
+      c = 3
+  [[tests.inputs]]
+    insert_at = "dedupe_field_order"
+    type = "log"
+    [tests.inputs.log_fields]
```

Comment:
Instead of using `log_fields` for the input, it might be better to use `raw`:

```toml
type = "raw"
value = "{\"a\": 1, \"b\": 2, \"c\": 3}"
```

This guarantees field order upon insert, whereas the TOML table fields are not guaranteed to preserve order.

Reply:
@binarylogic, it looks like when I try that the "value" winds up as a single string in the "message" field of the Event object, rather than as a broken down bunch of key-value pairs at the top level of the Event object.

Reply:
I wonder if I should just delete this behavior test if there's no good way to enforce field order.  It's already pretty much an exact dupe of a unit test that covers the same thing.

Reply:
I think it's fine for now, I would leave it. And you should be able to specify input in JSON form, right @Jeffail?